tools/pubsys/src/repo/check_expirations/mod.rs

```diff
@@ -0,0 +1,190 @@
+
+    #[structopt(long, parse(from_os_str))]
+    /// Path to root.json for this repo
+    root_role_path: PathBuf,
+
+    #[structopt(long, parse(try_from_str = parse_datetime))]
+    /// Finds metadata files expiring between now and a specified time; RFC3339 date or "in X hours/days/weeks"
+    expirations_between_now_and: Option<DateTime<Utc>>,
+}
+
+/// Checks for upcoming role expirations, gathers them in a list along with their expiration date.
+fn find_upcoming_metadata_expiration<T>(
+    repo: &Repository<'_, T>,
+    end_date: DateTime<Utc>,
+) -> HashMap<tough::schema::RoleType, DateTime<Utc>>
+where
+    T: tough::Transport,
+{
+    let mut expirations = HashMap::new();
+    info!(
+        "Looking for metadata expirations happening from now to {}",
+        end_date
+    );
+    if repo.root().signed.expires <= end_date {
+        expirations.insert(tough::schema::RoleType::Root, repo.root().signed.expires);
+    }
+    if repo.snapshot().signed.expires <= end_date {
+        expirations.insert(
+            tough::schema::RoleType::Snapshot,
+            repo.snapshot().signed.expires,
+        );
+    }
+    if repo.targets().signed.expires <= end_date {
+        expirations.insert(
+            tough::schema::RoleType::Targets,
+            repo.targets().signed.expires,
+        );
+    }
+    if repo.timestamp().signed.expires <= end_date {
+        expirations.insert(
+            tough::schema::RoleType::Timestamp,
+            repo.timestamp().signed.expires,
+        );
+    }
+
+    expirations
+}
+
+fn check_expirations(
+    transport: &RepoTransport,
+    root_role_path: &PathBuf,
+    metadata_url: &Url,
+    targets_url: &Url,
+    expirations_between_now_and: Option<DateTime<Utc>>,
+) -> Result<()> {
+    // Create a temporary directory where the TUF client can store metadata
+    let workdir = tempdir().context(repo_error::TempDir)?;
+    let settings = Settings {
+        root: File::open(root_role_path).context(repo_error::File {
+            path: root_role_path,
+        })?,
+        datastore: workdir.path(),
+        metadata_base_url: metadata_url.as_str(),
+        targets_base_url: targets_url.as_str(),
+        limits: Limits::default(),
+        // We're gonna check the expiration ourselves
+        expiration_enforcement: ExpirationEnforcement::Unsafe,
+    };
+
+    // Load the repository
+    let repo = Repository::load(transport, settings).context(repo_error::RepoLoad {
+        metadata_base_url: metadata_url.clone(),
+    })?;
+    info!("Loaded TUF repo:\t{}", metadata_url);
+
+    info!("Root expiration:\t{}", repo.root().signed.expires);
+    info!("Snapshot expiration:\t{}", repo.snapshot().signed.expires);
+    info!("Targets expiration:\t{}", repo.targets().signed.expires);
+    info!("Timestamp expiration:\t{}", repo.timestamp().signed.expires);
+    // Check for upcoming metadata expirations if a timeframe is specified
+    if let Some(expiration_limit) = expirations_between_now_and {
```

Comment:
If this subcommand is called without an expiration time. It just checks for expirations as of right now. Which is functionally the same thing as `validate-repo` but without downloading any targets.

Reply:
Well not exactly the same. `validate-repo` won't list out all the expired roles and just the first one where as in `check-repo-expirations` it will list out all of them.

Reply:
We have a default in the makefile, though.  Someone would have to call pubsys manually without a date to hit this case, right?  I don't think that's worth it..?